src/client/common/variables/environment.ts

```diff
@@ -16,10 +18,16 @@ export class EnvironmentVariablesService implements IEnvironmentVariablesService
         filePath?: string,
         baseVars?: EnvironmentVariables
     ): Promise<EnvironmentVariables | undefined> {
-        if (!filePath || !(await this.fs.localFileExists(filePath))) {
```

Comment:
Remove just this line of code saved almost 500-700ms.
We have two IO operations unnecessarily.